leshan-core/src/test/java/org/eclipse/leshan/core/node/codec/LwM2mNodeEncoderTest.java

```diff
@@ -316,5 +318,15 @@ public void senml_json_encode_resource() {
 
         String expected = "[{\"bn\":\"/3/0/0\",\"vs\":\"Open Mobile Alliance\"}]";
         Assert.assertEquals(expected, new String(encoded));
+
+        Map<Integer, Long> values = new HashMap<>();
+        values.put(0, 3800L);
+        values.put(1, 5000L);
+        oResource = LwM2mMultipleResource.newIntegerResource(7, values);
+        encoded = LwM2mNodeSenMLJsonEncoder.encode(oResource, new LwM2mPath("/3/0/7"), model,
+                new DefaultLwM2mValueConverter());
+
+        expected = "[{\"bn\":\"/3/0/7\",\"n\":\"0\",\"v\":3800},{\"n\":\"1\",\"v\":5000}]";
```

Comment:
I think the test is incorrect as `bn` should end with "/". (or n should start with "/")
I prefer you make 2 tests 1 for single resource and 1 for multiple resource.

Reply:
I checked [Table: 7.4.4.-5 JSON payload returned from example request to Device Object on Resource 0 of the LwM2M example client (Read /3/0/0)](http://www.openmobilealliance.org/release/LightweightM2M/V1_1-20180710-A/HTML-Version/OMA-TS-LightweightM2M_Core-V1_1-20180710-A.html#7-4-4-0-744-JSON), bn doesn't end with "/", if record is resouce.

_[{"bn":"/3/0/0","vs":"Open Mobile Alliance}]_

Reply:
You're correct about this example but the reason is not because this is a resource but because there is no "n".

In fact, you could put anything in bn and n, the only condition is that concatenation of both should be the path of the resource !

[SenML RFC](https://tools.ietf.org/html/rfc8428#section-2) says : _"To keep the messages small, it does not make sense to repeat the "n"
   field in each SenML Record, so there is a concept of a Base Name,
   which is simply a string that is prepended to the Name field of all
   elements in that Record and any Records that follow it. "_

Or look at field description of [Table: 7.4.4.-1 JSON format and description](http://www.openmobilealliance.org/release/LightweightM2M/V1_1-20180710-A/HTML-Version/OMA-TS-LightweightM2M_Core-V1_1-20180710-A.html#7-4-4-0-744-JSON)

You could also look at [LWM2M discussion](https://github.com/OpenMobileAlliance/OMA_LwM2M_for_Developers/issues/380) if you are not convinced !

Reply:
Understood. Thanks! 😉.